File: M3 Intermediate DSA/src/L4_Introduction_to_Arrays/Assignment2.java
package L4_Introduction_to_Arrays;

public class Assignment2 {

  public int solve(int[] A, int B) {
    int N = A.length;

    for (int index = 0; index < N - 1; index++) {
      for (int jIndex = index + 1; jIndex < N; jIndex++) {
        if (A[index] + A[jIndex] == B) {
          return 1;
        }
      }
    }

    return 0;
  }

}

File: M3_Intermediate_DSA/L2_Introduction_to_Arrays/assignment2.go
package main

func Solve2(A []int, B int) int {
	N := len(A)

	for index := 0; index < N-1; index++ {
		for jIndex := index + 1; jIndex < N; jIndex++ {
			if A[index]+A[jIndex] == B {
				return 1
			}
		}
	}

	return 0
}
